app/models/concerns/profile_fields.rb

```diff
@@ -47,5 +65,23 @@ def bm_versions
         upstream: false
       ).joins(:benchmark).pluck('benchmarks.version')
     end
+
+    def conflicting_relationships(rules, rule_groups)
+      conflicting_relationships = {}
+      RuleGroupRelationship.with_conflicts(rules).or(RuleGroupRelationship.with_conflicts(rule_groups)).each do |rgr|
+        conflicting_relationships[rgr.left] ||= []
+        conflicting_relationships[rgr.left] << rgr.right
+      end
+      conflicting_relationships
+    end
+
+    def required_relationships(rules, rule_groups)
+      required_relationships = {}
+      RuleGroupRelationship.with_requires(rules).or(RuleGroupRelationship.with_requires(rule_groups)).each do |rgr|
+        required_relationships[rgr.left] ||= []
+        required_relationships[rgr.left] << rgr.right
+      end
+      required_relationships
+    end
```

Comment:
This is repetition, what's the difference between the two methods? Can you please make it DRY?